### PR TITLE
[ML] Handle the "output memory allocator bytes" field

### PR DESCRIPTION
**Pull request by edsavage**
Handle the "output memory allocator bytes" field if and only if it is present in the model size stats, as reported by the C++ backend.

This new field reports how much memory is being used by the JSON parsing/writing routines in the `C++` code, and may be substantial.

This PR _must_ be merged prior to the corresponding `ml-cpp` one (https://github.com/elastic/ml-cpp/pull/2674), to keep CI tests happy.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatJobsAction.java [hunk 1]
```diff
@@ -213,6 +213,12 @@ protected Table getTableWithHeader(RestRequest request) {
                 .setAliases("mbaf", "modelBucketAllocationFailures")
                 .build()
         );
+        table.addCell(
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatJobsAction.java:216]: Nice, I'd forgotten about the cat jobs API

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Marking this as `backport pending` again because the backport was reverted.

**Comment by davidkyle** [comment]
#109847 addresses problems raised after the revert. I've removed the backport labels as this change will no longer be backported to 8.14.x.

**Pull request merged**